src/main/webapp/placeGuides/PlaceGuideOnList.js

```diff
@@ -136,27 +122,47 @@ class PlaceGuideOnList {
     return placeGuideNameContainer;
   }
 
-  foldedPlaceGuide_placeName(placeName) {
-    const placeNameElement = document.createElement('p');
-    placeNameElement.classList.add('mb-1');
-    placeNameElement.innerText = placeName;
-    return placeNameElement;
-  }
-
   foldedPlaceGuide_buttons(placeGuideId, audioKey) {
     const buttonsContainer = document.createElement('div');
     buttonsContainer.classList.add('mdc-card__action-icons');
     this.createAudioButton(audioKey, buttonsContainer);
     const expandButton = this.getPlaceGuideButtonWithPreparedClasses();
     expandButton.setAttribute('title', 'expand');
     expandButton.innerText = 'open_in_full';
+    const placeGuideLocation = this._placeGuideProperties.location;
     expandButton.addEventListener('click', function() {
-      PlaceGuideOnList.expand(placeGuideId);
+      PlaceGuideOnList.expand(placeGuideId, placeGuideLocation);
     });
     buttonsContainer.appendChild(expandButton);
     return buttonsContainer;
   }
 
+  static expand(placeGuideId, location) {
+    const divId = 'placeGuideOnList-' + '{' + placeGuideId + '}';
+    const placeGuideDiv = document.getElementById(divId);
+    const placeNameDiv = document.getElementById(
+        PlaceGuideOnList.getPlaceGuidePlaceNameElementId(placeGuideId));
+    if (placeNameDiv !== null && placeNameDiv.innerText === "") {
```

Comment:
Is this null check needed?

Reply:
Yes, because the div for the placename is created only if the placeguide has a placeId(i.e. it is linked to a place to to a point defined by coordinates).